### PR TITLE
TW-879: Redesign Chat Invitation Dialog

### PR DESCRIPTION
**Pull request by imGok**
#879 

Mobile Reject

https://github.com/linagora/twake-on-matrix/assets/48354990/6efa7b94-6223-4814-ba5d-c6eb48c13424

Mobile Accept

https://github.com/linagora/twake-on-matrix/assets/48354990/66c3f1f8-6dee-4256-9d3b-71654781a8a4

Web Reject

https://github.com/linagora/twake-on-matrix/assets/48354990/b816be8b-7296-4e3c-9c9b-cf723fcc1c8a

Web Accept

https://github.com/linagora/twake-on-matrix/assets/48354990/a802fed7-85f7-46f3-af24-14b961e942d3

### CODE REVIEW

File: lib/pages/chat/chat_invitation_body.dart [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+  final ChatController controller;
+
+  const ChatInvitationBody(this.controller, {Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    controller.matrix ??= Matrix.of(context);
+    final client = controller.matrix!.client;
+    controller.sendingClient ??= client;
+    controller.room = controller.sendingClient!.getRoomById(controller.roomId!);
```
Comment (sherlockvn) [inline lib/pages/chat/chat_invitation_body.dart:21]: Put what you need in the constructor, don't put the whole controller in constructor, it's hard to reuse

File: lib/pages/chat/chat_view_body.dart [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+  final ChatController controller;
+
+  const ChatViewBody(this.controller, {Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    controller.matrix ??= Matrix.of(context);
+    final client = controller.matrix!.client;
+    controller.sendingClient ??= client;
+    controller.room = controller.sendingClient!.getRoomById(controller.roomId!);
```
Comment (sherlockvn) [inline lib/pages/chat/chat_view_body.dart:29]: same

**Comment by nqhhdev** [comment]
- IMO: Dialog is full screen, pls update it
Screenshot 2023-11-14 at 12 11 00

File: lib/pages/chat/chat_invitation_body.dart (hunk 1 above)
Comment (nqhhdev) [reply lib/pages/chat/chat_invitation_body.dart:21]: I agree with @sherlockvn, create some properties for ti

File: lib/pages/chat/chat_invitation_body.dart [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+
+  String _getDisplayInviterName() {
+    if (controller.room!.isDirectChat) {
+      return controller.room!.getLocalizedDisplayname();
+    } else {
+      return controller
+              .room!.lastEvent?.senderFromMemoryOrFallback.displayName ??
+          controller.room!.getLocalizedDisplayname();
+    }
+  }
```
Comment (nqhhdev) [inline lib/pages/chat/chat_invitation_body.dart:165]: IMO: Why don't you write inside ChatController?

File: lib/pages/chat/chat_view_body.dart [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+    // if (controller.room == null) {
+    //   return const SizedBox.shrink();
+    // }
+
+    final bottomSheetPadding = TwakeThemes.isColumnMode(context) ? 16.0 : 8.0;
```
Comment (nqhhdev) [inline lib/pages/chat/chat_view_body.dart:34]: move it to style

File: lib/pages/chat/chat_view_body.dart [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+  //             L10n.of(context)!.youNeedToAcceptTheInvitation,
+  //             style: Theme.of(context).textTheme.bodyLarge?.copyWith(
+  //                   color: LinagoraRefColors.material().tertiary[30],
+  //                 ),
+  //           ),
+  //         ],
+  //       ),
+  //     ),
+  //   );
+  // }
```
Comment (nqhhdev) [inline lib/pages/chat/chat_view_body.dart:205]: Remove it?

**Comment by hoangdat** [comment]
@imGok , please use dialog in full screen. Why? Reason: https://github.com/linagora/twake-on-matrix/issues/866

File: lib/pages/chat/chat_invitation_body.dart (hunk 1 above)
Comment (hoangdat) [reply lib/pages/chat/chat_invitation_body.dart:21]: Totally agree, do any logic in `build()` is not a smart choice

File: lib/pages/chat/chat_invitation_body.dart (hunk 3 above)
Comment (imGok) [reply lib/pages/chat/chat_invitation_body.dart]: Done it :)

File: lib/pages/chat/chat_view_body.dart (hunk 5 above)
Comment (imGok) [reply lib/pages/chat/chat_view_body.dart]: Done it :)

File: lib/pages/chat/chat_view_body.dart (hunk 2 above)
Comment (imGok) [reply lib/pages/chat/chat_view_body.dart]: Done it :)

**Comment by imGok** [comment]
image
Full screen modal

File: lib/pages/chat/chat_view.dart [hunk 6]
```diff
@@ -124,16 +114,11 @@ class ChatView extends StatelessWidget with MessageContentMixin {
 
   @override
   Widget build(BuildContext context) {
-    controller.matrix ??= Matrix.of(context);
-    final client = controller.matrix!.client;
-    controller.sendingClient ??= client;
-    controller.room = controller.sendingClient!.getRoomById(controller.roomId!);
+    controller.initRoom();
```
Comment (hoangdat) [inline lib/pages/chat/chat_view.dart:117]: we can not do it in better way, keep it as current status.

File: lib/pages/chat/chat.dart [hunk 7]
```diff
@@ -1613,6 +1563,52 @@ class ChatController extends State<Chat>
...
 
+  void onAcceptInvitation() async {
+    await TwakeDialog.showFutureLoadingDialogFullScreen(
+      future: () async {
+        final waitForRoom = room?.client.waitForRoomInSync(
+          room!.id,
+          join: true,
+        );
+        await room?.join();
+        await waitForRoom;
```
[the remark this answers is not in the document]
Comment (imGok) [reply lib/pages/chat/chat.dart:1574]: It's the same logic as before I didn't change anything, @nqhhdev can give you more details about it

File: lib/pages/chat/chat_view.dart (hunk 6 above)
Comment (imGok) [reply lib/pages/chat/chat_view.dart]: reverted

**Review by sherlockvn — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Review by nqhhdev — APPROVED** [review]

**Pull request merged**